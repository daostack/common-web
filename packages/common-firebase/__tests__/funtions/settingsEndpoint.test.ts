import supertest from 'supertest';

import { runTest } from '@helpers/runTest';
import { env } from '@env';

import "@functions";


runTest((funcs) => {
  const graphql = supertest(funcs.graphql);

  describe('Settings endpoint', () => {
    it('should be able to get settings endpoint', async () => {
      const res = await graphql.get('/settings');

      expect(res.body.message).toMatchSnapshot();
      expect(res.status).toBe(200);
    });

    it('should match current settings',  async () => {
      const res = await graphql.get('/settings');


      expect(res.body.currentEnvironment).toMatch(env.environment);
      expect(res.body.graphql.subgraph).toBe(env.graphql.subgraphName);
    });
  });
});
