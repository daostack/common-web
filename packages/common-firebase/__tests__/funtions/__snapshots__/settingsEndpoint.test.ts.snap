// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cloud Functions Settings endpoint should be able to get settings endpoint 1`] = `"Setting successfully acquired!"`;
